torch/_refs/__init__.py

```diff
@@ -2684,8 +2684,28 @@ def flipud(a: TensorLikeType) -> TensorLikeType:
 
 
 # CompositeImplicitAutograd - don't register decomp
-def narrow(a: TensorLikeType, dim: int, start: int, length: int) -> TensorLikeType:
+def narrow(
+    a: TensorLikeType, dim: int, start: Union[int, TensorLikeType], length: int
+) -> TensorLikeType:
+    # Support Tensor overload that was added for XLA:
+    # https://github.com/pytorch/pytorch/issues/31558
+    if isinstance(start, TensorLike):
+        check(
+            start.dim() == 0 and utils.is_integer_dtype(start.dtype),
+            lambda: "start must be an 0-dim integral Tensor.",
+        )
+        start = start.item()  # type: ignore[assignment]
+    check(a.dim() > 0, lambda: "narrow() cannot be applied to a 0-dim tensor.")
     dim = utils.canonicalize_dim(a.ndim, dim)
+    cur_size = a.size(dim)
```

Comment:
Nit: to be more consistent with our language elsewhere, this would be the length of the dimension (which is confusing because the call to acquire the length is `size`, but whatever), so the variable would be `cur_length`